mmtk/src/api.rs

```diff
@@ -29,12 +30,22 @@ pub extern "C" fn jikesrvm_gc_init(jtoc: *mut c_void, heap_size: usize) {
         JTOC_BASE = Address::from_mut_ptr(jtoc);
         BOOT_THREAD = OpaquePointer::from_address(VMCollection::thread_from_id(1));
     }
-    // MMTk should not be used before gc_init, and gc_init is single threaded. It is fine we get a mutable reference from the singleton.
-    #[allow(clippy::cast_ref_to_mut)]
-    let singleton_mut =
-        unsafe { &mut *(&*SINGLETON as *const MMTK<JikesRVM> as *mut MMTK<JikesRVM>) };
-    memory_manager::gc_init(singleton_mut, heap_size);
-    debug_assert!(731 == JikesRVM::mm_entrypoint_test(21, 34, 9, 8));
+
+    // set heap size
+    memory_manager::process(&BUILDER, "heap_size", heap_size.to_string().as_str());
```

Comment:
If mmtk-core can expose `&mut Options` directly, we won't need to convert `heap_size` into a string.